core/rawdb/freezer.go

```diff
@@ -108,7 +108,13 @@ func NewFreezer(datadir string, namespace string, readonly bool, maxTableSize ui
 	// Leveldb uses LOCK as the filelock filename. To prevent the
 	// name collision, we use FLOCK as the lock name.
 	lock := flock.New(flockFile)
-	if locked, err := lock.TryLock(); err != nil {
+	tryLock := func() (bool, error) {
+		if readonly {
+			return lock.TryRLock()
+		}
+		return lock.TryLock()
+	}
+	if locked, err := tryLock(); err != nil {
```

Comment:
It might be that this works: 
```suggestion
	tryLock := lock.TryLock
	if readonly {
		tryLock = lock.TryRLock
	}
	if locked, err := tryLock(); err != nil {
```